api.php

```diff
@@ -47,6 +48,58 @@
         $data = array_merge($data, getAllQueries());
     }
 
+    if (isset($_GET['getVersions'])) {
+        $data = array_merge($data, getVersions());
+    }
+
+    if (isset($_GET['getStatus'])) {
+        $status['status'] = getStatus();
+        $data = array_merge($data, $status);
+    }
+
+    if (isset($_GET['getToken'])) {
+        if(empty($_SESSION['token'])) {
+            $_SESSION['token'] = base64_encode(openssl_random_pseudo_bytes(32));
+        }
+        $token['token'] = $_SESSION['token'];
+        $data = array_merge($data, $token);
+    }
```

Comment:
This is going to have to be changed in some way... Anyone from any origin can take the token and use it. [This](https://stormpath.com/blog/angular-xsrf) looks like it could help.